components/fxa-client/src/migrator.rs

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{error::*, scoped_keys::ScopedKey, scopes, FirefoxAccount};
+
+impl FirefoxAccount {
+    /// Migrate from a logged-in browserid Firefox Account.
+    ///
+    /// * `session_token` - Hex-formatted session token.
+    /// * `k_xcs` - Hex-formatted kXCS.
+    /// * `k_sync` - Hex-formatted kSync.
+    ///
+    /// **💾 This method alters the persisted account state.**
+    pub fn migrate_from_session_token(
+        &mut self,
+        session_token: &str,
+        k_sync: &str,
+        k_xcs: &str,
+    ) -> Result<()> {
+        // Trade our session token for a refresh token.
```

Comment:
Should this error out if `self.stage.sessionToken` is already `Some`? It almost certainly indicates a logic error on behalf of the caller.

Reply:
I think we can error out for now